modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java

```diff
@@ -233,6 +252,23 @@ private ClassLoaderModelBuilder createBuilderWithoutExportedPackages(ClassLoader
     return classLoaderModelBuilder;
   }
 
+  private ClassLoaderModelBuilder createBuilderWithoutDependency(ClassLoaderModel originalClassLoaderModel,
+                                                                 BundleDependency dependencyToBeExcluded) {
+    ClassLoaderModelBuilder classLoaderModelBuilder = new ClassLoaderModelBuilder()
+        .dependingOn(originalClassLoaderModel.getDependencies().stream()
+            .filter(dependency -> !dependency.equals(dependencyToBeExcluded)).collect(
+                                                                                      Collectors
+                                                                                          .toCollection(() -> new LinkedHashSet<>())))
```

Comment:
why not just `toSet()` collector?

Reply:
Order matters here, as later these dependencies will end up creating the classloaders for each plugin and later the extension models are generated for each plugin (classloader) and it is expected that the plugins list is sorted by dependencies as extension models require context to resolve things (for instance: http cannot be loaded if extension model for sockets was not loaded before).